resources/BUILD.gn

```diff
@@ -0,0 +1,176 @@
+  ]
+  output = "$root_out_dir/xwalk_100_percent.pak"
+
+  if (toolkit_views) {
+    deps += [ "//ui/views/resources:resources_grd_grit" ]
+    sources +=
+        [ "$root_gen_dir/ui/views/resources/views_resources_100_percent.pak" ]
+  }
+}
+
+# GYP version: xwalk.gyp:xwalk_resources_200_percent_pak
+repack("xwalk_resources_200_percent_pak") {
+  deps = [
+    "//components/resources",
+    "//content/app/resources:resources_grit",
+    "//third_party/WebKit/public:image_resources_grit",
+    "//ui/resources:resources",
+  ]
+  sources = [
+    "$root_gen_dir/blink/public/resources/blink_image_resources_200_percent.pak",
+    "$root_gen_dir/components/components_resources_200_percent.pak",
+    "$root_gen_dir/content/app/resources/content_resources_200_percent.pak",
+    "$root_gen_dir/ui/resources/ui_resources_200_percent.pak",
+  ]
+  output = "$root_out_dir/xwalk_200_percent.pak"
+
+  if (toolkit_views) {
+    deps += [ "//ui/views/resources:resources_grd_grit" ]
+    sources +=
+        [ "$root_gen_dir/ui/views/resources/views_resources_200_percent.pak" ]
+  }
+}
+
+# GYP version: xwalk.gyp:xwalk_resources_300_percent_pak
+repack("xwalk_resources_300_percent_pak") {
+  deps = [
+    "//components/resources",
+    "//ui/resources:resources",
+  ]
+  sources = [
+    "$root_gen_dir/components/components_resources_300_percent.pak",
+    "$root_gen_dir/ui/resources/ui_resources_300_percent.pak",
+  ]
+  output = "$root_out_dir/xwalk_300_percent.pak"
+
+  if (toolkit_views) {
+    deps += [ "//ui/views/resources:resources_grd_grit" ]
+    sources +=
+        [ "$root_gen_dir/ui/views/resources/views_resources_300_percent.pak" ]
+  }
+}
+
+# GYP version: xwalk.gyp:generate_xwalk_resources
+grit("generate_xwalk_resources") {
+  source = "//xwalk/runtime/resources/xwalk_resources.grd"
+  resource_ids = "//xwalk/resources/resource_ids"
+  outputs = [
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources.h",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.cc",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.h",
+    "$root_gen_dir/xwalk/resources/xwalk_resources.pak",
+    "$root_gen_dir/xwalk/resources/xwalk_resources.rc",
+  ]
+}
+
+copy("xwalk_resources") {
+  deps = [
+    ":generate_xwalk_resources",
+  ]
+  sources = [
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources.h",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.cc",
+    "$root_gen_dir/xwalk/resources/grit/xwalk_resources_map.h",
+  ]
+  outputs = [
+    "$root_gen_dir/xwalk/grit/{{source_file_part}}",
+  ]
+}
+
+grit("generate_xwalk_strings") {
+  source = "//xwalk/runtime/resources/xwalk_strings.grd"
+  resource_ids = "//xwalk/resources/resource_ids"
+  output_dir = "$root_gen_dir/xwalk/locales"
+  outputs = [
+    "$root_gen_dir/xwalk/grit/xwalk_strings.h",
+    "$root_gen_dir/xwalk/locales/en-US.pak",
+    "$root_gen_dir/xwalk/locales/zh-CN.pak",
+  ]
+}
+
+copy("xwalk_strings") {
```

Comment:
I delete the out/ folder and build again. no error found.  But I did meet this before: in xwalk.gyp:
```
  ['OS=="linux" and use_ozone!=1', {                                                                    
            'defines': ['USE_GTK_UI'],
            'dependencies': [
              '../chrome/browser/ui/libgtk2ui/libgtk2ui.gyp:gtk2ui',
            ],
  }]
```
If add the `libgtk2ui` in your deps, you will find the` //chrome/BUILD.gn` is finally imported, then it also try to generate the output file `"locales/en-US.pak"`. That's why it complains collisions.
Peter checked the "USE_GTK_UI" usage in crosswalk,  we think it should not be used anymore. Without build this libgtk2ui, there is not collision error.
Actually I'm not understand this background of `USE_GTK_UI` and `libgtk2ui`,   @PeterWangIntel  Any comments on this?